assets/eip-7092/ERC7092.sol

```diff
@@ -28,125 +25,110 @@ contract ERC7092 is IERC7092, BondStorage {
+    *         ex: If the denomination is equal to 1,000 and the currency is USDC, then the bond denomination is equal to 1,000 USDC
+    */
+    function denomination() external view returns(uint256) {
+        return _bonds[bondISIN].denomination;
     }
 
+    /**
+    * @notice Returns the issue volume (total debt amount). It is RECOMMENDED to express the issue volume in denomination unit.
+    */
     function issueVolume() external view returns(uint256) {
-        return _bond[bondISIN].issueVolume;
+        return _bonds[bondISIN].issueVolume;
     }
 
-    function couponRate() external view returns(uint256) {
-        return _bond[bondISIN].couponRate;
+    /**
+    * @notice Returns the bond tokens total supply
+    */
+    function totalSupply() external view returns(uint256) {
+        return _bonds[bondISIN].issueVolume / _bonds[bondISIN].denomination;
     }
 
-    function couponType() external view returns(uint256) {
-        return _bond[bondISIN].couponType;
-    }
-
-    function couponFrequency() external view returns(uint256) {
-        return _bond[bondISIN].couponFrequency;
+    /**
+    * @notice Returns the bond interest rate. It is RECOMMENDED to express the interest rate in basis point unit.
+    *         1 basis point = 0.01% = 0.0001
+    *         ex: if interest rate = 5%, then coupon() => 500 basis points
+    */
+    function couponRate() external view returns(uint256) {
+        return _bonds[bondISIN].couponRate;
     }
 
+    /**
+    * @notice Returns the date when bonds were issued to investors. This is a Unix Timestamp like the one returned by block.timestamp
+    */
     function issueDate() external view returns(uint256) {
-        return _bond[bondISIN].issueDate;
-    }
-
-    function maturityDate() public view returns(uint256) {
-        return _bond[bondISIN].maturityDate;
+        return _bonds[bondISIN].issueDate;
     }
 
-    function dayCountBasis() external view returns(uint256) {
-        return _bond[bondISIN].dayCountBasis;
+    /**
+    * @notice Returns the bond maturity date, i.e, the date when the pricipal is repaid. This is a Unix Timestamp like the one returned by block.timestamp
+    *         The maturity date MUST be greater than the issue date
+    */
+    function maturityDate() external view returns(uint256) {
+        return _bonds[bondISIN].maturityDate;
     }
 
+    /**
+    * @notice Returns the principal of an account. It is RECOMMENDED to express the principal in the bond currency unit (USDC, DAI, etc...)
+    * @param _account account address
+    */
     function principalOf(address _account) external view returns(uint256) {
         return _principals[_account];
     }
 
-    function balanceOf(address _account) public view returns(uint256) {
-        require(_bondStatus == BondStatus.ISSUED, "ERC7092: NOT_ISSUED_OR_REDEEMED");
-
-        return _principals[_account] / _bond[bondISIN].denomination;
-    }
-
-    function totalSupply() public view returns(uint256) {
```

Comment:
@Pandapip1,

The `totalSupply` function is removed from the standard since it can be calculated from the `issueVolume` and the `denomination`. But has it is specified in the EIP, it is RECOMMENDED to implement the totalSupply function in any contract that implements the ERC-7092